src/main/kotlin/hr/foi/pop/backend/services/UserService.kt

```diff
@@ -41,6 +44,26 @@ class UserService {
         return user
     }
 
+    @Throws(ActivateUserException::class)
```

Comment:
No need for this annotation. It should be caught automatically in the ControllerAdvice class, so the user of this method has no need to know about this (unless he's interested in what happens when this method is misused).

Reply:
Should I then put all exceptions inside GlobalExceptionHandler?

Reply:
Erm... We might be talking about different things here. This @Throws anntoation is only for Java interoperability on checked exceptions. So, it has nothing with handling of exceptions.
As far as global handling is concerned - no, I don't think exceptions such as ActivateUserException should be globally caught, but rather in the controller.